### PR TITLE
feat: uplift and modernise significantly

### PR DESCRIPTION
**Pull request by mble-sfdc**
This PR covers a large uplift to the demo app, modernising it and hopefully making it more straightforward to understand.

The highlights are:
- `bsm/sarama-cluster` is removed, as a high level group consumer is now included in `IBM/sarama`.
- Tests and associated workflows are added.
- All dependencies are updated to their latest versions.
- Config and Kafka setup are moved into their own internal packages.
- HTML template and associated Javascript have been modernised, using the latest Bootstrap version from a CDN, and removing the jQuery dependency opting for native builtins, including `fetch`.
- Readme updated to reflect correct trademark usage of Apache Kafka.

### CODE REVIEW

**Review by dhagberg-sf — APPROVED** [review]
Couple questions but looks good - great stuff - much success.

File: internal/transport/transport.go [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+	return &KafkaClient{
+		AsyncProducer: asyncProducer,
+		Producer:      producer,
+		Consumer:      consumer,
+	}, nil
+}
+
+// Close closes the KafkaClient
+func (kc *KafkaClient) Close() {
+	kc.Producer.Close()
```
[the remark this answers is not in the document]
Comment (mble-sfdc) [reply internal/transport/transport.go:241]: Ugh yes.

File: internal/transport/transport.go [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+				"error consuming",
+				"err", err,
+			)
+		}
+
+		if ctx.Err() != nil {
+			return
+		}
+
+		handler.Ready = make(chan bool)
```
[the remark this answers is not in the document]
Comment (mble-sfdc) [reply internal/transport/transport.go:315]: Oops, yes. This is meant to be outside the loop - a refactoring gone wrong.

File: internal/transport/transport.go [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+				return
+			}
+
+			slog.Error(
+				"error consuming",
+				"err", err,
+			)
+		}
+
+		if ctx.Err() != nil {
```
[the remark this answers is not in the document]
Comment (mble-sfdc) [reply internal/transport/transport.go:311]: No need - that's that `ctx.Err()` does:
>If Done is not yet closed, Err returns nil.
> If Done is closed, Err returns a non-nil error explaining why:
> Canceled if the context was canceled or DeadlineExceeded if the context's deadline passed.
> After Err returns a non-nil error, successive calls to Err return the same error.

File: internal/transport/transport.go [hunk 4]
```diff
@@ -0,0 +1,359 @@
...
+
+	consumer, err := sarama.NewConsumerGroup(ac.BrokerAddresses(), ac.Group(), kafkaConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	return consumer, nil
+}
+
+// KafkaClient is a wrapper around a Sarama SyncProducer and ConsumerGroup
```
[the remark this answers is not in the document]
Comment (mble-sfdc) [reply internal/transport/transport.go]: Resolved

File: internal/transport/transport.go [hunk 5]
```diff
@@ -0,0 +1,359 @@
...
+	c.JSON(http.StatusOK, mb.receivedMessages)
+}
+
+// SaveMessage saves a message to the buffer
+func (mb *MessageBuffer) SaveMessage(msg Message) {
+	mb.ml.Lock()
+	defer mb.ml.Unlock()
+
+	if len(mb.receivedMessages) >= mb.MaxSize {
+		mb.receivedMessages = mb.receivedMessages[1:]
```
[the remark this answers is not in the document]
Comment (mble-sfdc) [reply internal/transport/transport.go:57]: Resolved.

File: internal/transport/transport.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (dhagberg-sf) [reply internal/transport/transport.go:313]: 👍  Thanks for checking through these...

File: internal/transport/transport.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (dhagberg-sf) [reply internal/transport/transport.go:317]: Acknowledged.  Will give some thought if there's a better way to do this but should not block this hunka-merge.

**Review by dhagberg-sf — APPROVED** [review]
re- ✅

**Pull request merged**